sds_data_manager/lambda_code/IAlirtCode/ialirt_data_query_api.py

```diff
@@ -0,0 +1,249 @@
+"""I-ALiRT Data Query lambda."""
+
+import json
+import logging
+import os
+import time
+from datetime import datetime, timedelta, timezone
+from urllib.parse import quote_plus, unquote_plus, urlencode
+
+import boto3
+from boto3.dynamodb.conditions import Key
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+table_name = os.environ.get("DATA_TABLE")
+region = os.environ.get("AWS_DEFAULT_REGION", "us-west-2")
+dynamodb = boto3.resource("dynamodb", region_name=region)
+table = dynamodb.Table(table_name)
+
+
+def apply_time_filters(params: dict, query_kwargs: dict) -> Key:
+    """Apply the filters for time.
+
+    Parameters
+    ----------
+    params : dict
+        Event parameters.
+    query_kwargs : dict
+        Query keyword arguments.
+
+    Returns
+    -------
+    key_expr : Key
+        The updated key expression with time filters applied.
+    """
+    key_expr = query_kwargs["KeyConditionExpression"]
+
+    start = params.get("time_utc_start") or params.get("met_in_utc_start")
+    end = params.get("time_utc_end") or params.get("met_in_utc_end")
+
+    if start and end:
+        key_expr &= Key("time_utc").between(start, end)
+    elif start:
+        key_expr &= Key("time_utc").gte(start)
+    else:
+        return _error(400, "End time provided without start time")
+
+    query_kwargs["KeyConditionExpression"] = key_expr
+
+    return key_expr
```

Comment:
return the dictionary that has been modified here instead? Otherwise it has been modified at a distance.
```suggestion
    return query_kwargs
```